.github/workflows/benchmark-action.yaml

```diff
@@ -0,0 +1,76 @@
+name: benchmarks
+
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    types: [opened, reopened, synchronize]
+  workflow_dispatch:
+
+permissions:
+  # deployments permission to deploy GitHub pages website
+  deployments: write
+  # contents permission to update benchmark contents in gh-pages branch
+  contents: write
+
+
+# see https://github.com/benchmark-action/github-action-benchmark
+jobs:
+  benchmark:
+    name: Performance regression check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v1 # default version of go is 1.10
+        with:
+          go-version: "1.17.6"
+      - name: Install Carvel Tools
+        run: ./hack/install-deps.sh
+      # Run benchmark with `go test -bench` and stores the output to a file
+      - name: Setup k8s and Run benchmark
+        run: |
+          set -e -x
+          mkdir /tmp/bin
+          export PATH=/tmp/bin:$PATH
+          curl -sLo /tmp/bin/minikube https://github.com/kubernetes/minikube/releases/latest/download/minikube-linux-amd64
+          chmod +x /tmp/bin/minikube
+          minikube start --driver=docker
+          eval $(minikube docker-env --shell=bash)
+
+          ./hack/deploy-test.sh
+
+          go test ./test/bench/... -bench=. | tee output.txt
+      # Download previous benchmark result from cache (if exists)
+      - name: Download previous benchmark data
+        uses: actions/cache@v1
+        with:
+          path: ./cache
+          key: ${{ runner.os }}-benchmark
+      # Run `github-action-benchmark` action
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
```

Comment:
since this is an external action (and it writes to our repo), i would recommend locking it down by sha.

Reply:
just a note that to test this and ensure it's working i commented out the branch conditional so it'll need another commit/push before its ready to merge.  i'll ping and resolve this conversation when it's done.

Reply:
confirmed I pinned to hashes correctly and restored the conditionals.